### ISSUE TITLE
Change the baseline from 2 to 10 as described in the docs for `cyclomatic_complexity` lint

### ISSUE THREAD

**Issue by arthurbcd** [solid-software/solid_lints#146]
In the docs, you say the baseline is 10:

```md
cyclomatic_complexity

According to the reference, we use 10 as the baseline value.

Reference: NIST 500-235 item 2.5
```

Reference: https://lints.solid.software/docs/rulesets/main

But in the actual code, we find this:

```dart
/// Cyclomatic complexity metric limits configuration.
class CyclomaticComplexityParameters {
  /// Threshold cyclomatic complexity level, exceeding it triggers a warning.
  final int maxComplexity;

  static const _defaultMaxComplexity = 2; // <- Why it's 2?

  /// Constructor for [CyclomaticComplexityParameters] model
  const CyclomaticComplexityParameters({
    required this.maxComplexity,
  });

  /// Method for creating from json data
  factory CyclomaticComplexityParameters.fromJson(Map<String, Object?> json) =>
      CyclomaticComplexityParameters(
        maxComplexity: json['max_complexity'] as int? ?? _defaultMaxComplexity,
      );
}
```

**Comment by illia-romanenko** [comment]
Hi @arthurbcd, this is an interesting observation. I think that 2 was used by default there, and it might not be very reasonable in the end.

Assumption is that our bundled analysis options will be used as it contains other configuration rules that are based on standards where `cyclomatic_complexity` it's set to 10: https://github.com/solid-software/solid_lints/blob/383581ee8dbbd09e52915d52cd821bd27f5293c5/lib/analysis_options.yaml#L55

<img width="351" alt="image" src="https://github.com/solid-software/solid_lints/assets/442086/d245e234-33e2-4f47-9ebf-696b5f06378a">

**Comment by illia-romanenko** [comment]
Closed in the latest commit.
